dev/io.openliberty.mcp.internal/src/io/openliberty/mcp/internal/McpServlet.java

```diff
@@ -96,6 +99,18 @@ protected void doPost(HttpServletRequest req, HttpServletResponse resp) throws S
         McpTransport transport = new McpTransport(req, resp, jsonb);
         try {
             transport.init();
+
+            RequestMethod method = transport.getMcpRequest().getRequestMethod();
+            if (method != RequestMethod.INITIALIZE) {
```

Comment:
I think `ping` is also permitted without a session id. This will also require a test.